src/main/java/com/zaxxer/nuprocess/linux/ProcessEpoll.java

```diff
@@ -160,22 +138,19 @@ public void queueWrite(LinuxProcess process)
          if (stdin == -1) {
            return;
          }
-         EpollEvent event = eventPool.take();
+
+         EpollEvent event = process.getEpollEvent();
          event.setEvents(LibEpoll.EPOLLOUT | LibEpoll.EPOLLONESHOT | LibEpoll.EPOLLRDHUP | LibEpoll.EPOLLHUP);
          event.setFileDescriptor(stdin);
          int rc = LibEpoll.epoll_ctl(epoll, LibEpoll.EPOLL_CTL_MOD, stdin, event.getPointer());
          if (rc == -1) {
             LibEpoll.epoll_ctl(epoll, LibEpoll.EPOLL_CTL_DEL, stdin, event.getPointer());
             rc = LibEpoll.epoll_ctl(epoll, LibEpoll.EPOLL_CTL_ADD, stdin, event.getPointer());
+            if (rc == -1) {
+               int errno = Native.getLastError();
+               throw new RuntimeException("Unable to register new event to epoll queue, errno: " + errno);
```

Comment:
This change to add `errno` values stems from some debugging issues we had with errors during `closeStdin` calls in Bitbucket Server. If a process racily terminates while we're trying to close `stdin`, this exception ends up thrown with `EBADF` (9). I'll try to work on a separate patch that will actually handle that case cleanly, but for the moment this at least makes it possible to identify the case where closing `stdin` raced with process termination.